internal/uvm/pipes.go

```diff
@@ -12,6 +12,32 @@ import (
 
 const pipePrefix = `\\.\pipe\`
 
+// PipeMount contains the host path for pipe mount
+type PipeMount struct {
```

Comment:
We should store the vm the resource is associated to as part of the pipe struct. Then we can just have a `Close()` or more accurately a Release() method of the resource. Then the caller just has to loop all `Resources` and call `Release()` on them.

Reply:
And generically we should do this pattern for all resource types.

Reply:
@jterry75 This is an approach me, Kevin and John discussed but we decided to go with another way, but this can also work. I'd like some more feedback from others before I'd start to refactor because it seems like we have conflicting opinions on whether to embed the UVM in all of the resource structs. Both are good options. The networking and layers side of things complicate it as they rely on a bunch of conditionals for their cleanup but as they arent following an interface right now it doesnt really matter. Are you suggesting to go with them implementing a Resource interface so they all have the same parameters? If so layers needs a way to access the "all" bool and the "containerRootInUVM" path.